kubernetes/structure_persistent_volume_spec.go

```diff
@@ -1028,6 +1034,27 @@ func patchPersistentVolumeSpec(pathPrefix, prefix string, d *schema.ResourceData
 			})
 		}
 	}
+	if d.HasChange(prefix + "node_affinity") {
```

Comment:
There is a bit of a conflict here.
The schema sets all the sub-attributes of `node_selector_term` as `ForceNew`. This negates the need for patch operations to be implemented for them, like you have done here.

Here's the result of a quick test I did trying to modify `node_selector_term`. I cannot get the plan to show a modify (update) operation as it's always trying to replace the resource because of the `ForceNew` flags.

```
------------------------------------------------------------------------

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
-/+ destroy and then create replacement

Terraform will perform the following actions:

-/+ kubernetes_persistent_volume.test (new resource required)
      id:                                                                                           "test-pv-affinity" => <computed> (forces new resource)
      metadata.#:                                                                                   "1" => "1"
      metadata.0.generation:                                                                        "0" => <computed>
      metadata.0.name:                                                                              "test-pv-affinity" => "test-pv-affinity"
      metadata.0.resource_version:                                                                  "2975" => <computed>
      metadata.0.self_link:                                                                         "/api/v1/persistentvolumes/test-pv-affinity" => <computed>
      metadata.0.uid:                                                                               "ef41d59b-fe8f-11e8-9f90-000c2979c097" => <computed>
      spec.#:                                                                                       "1" => "1"
      spec.0.access_modes.#:                                                                        "1" => "1"
      spec.0.access_modes.1254135962:                                                               "ReadWriteMany" => "ReadWriteMany"
      spec.0.capacity.%:                                                                            "1" => "1"
      spec.0.capacity.storage:                                                                      "10Mi" => "10Mi"
      spec.0.node_affinity.#:                                                                       "1" => "1"
      spec.0.node_affinity.0.required.#:                                                            "1" => "1"
      spec.0.node_affinity.0.required.0.node_selector_term.#:                                       "1" => "1"
      spec.0.node_affinity.0.required.0.node_selector_term.0.match_expressions.#:                   "1" => "2" (forces new resource)
      spec.0.node_affinity.0.required.0.node_selector_term.0.match_expressions.0.key:               "hostname" => "hostname"
      spec.0.node_affinity.0.required.0.node_selector_term.0.match_expressions.0.operator:          "In" => "In"
      spec.0.node_affinity.0.required.0.node_selector_term.0.match_expressions.0.values.#:          "1" => "1"
      spec.0.node_affinity.0.required.0.node_selector_term.0.match_expressions.0.values.2666930069: "foobar" => "foobar"
      spec.0.node_affinity.0.required.0.node_selector_term.0.match_expressions.1.key:               "" => "kubernetes.io/hostname" (forces new resource)
      spec.0.node_affinity.0.required.0.node_selector_term.0.match_expressions.1.operator:          "" => "In" (forces new resource)
      spec.0.node_affinity.0.required.0.node_selector_term.0.match_expressions.1.values.#:          "" => "1" (forces new resource)
      spec.0.node_affinity.0.required.0.node_selector_term.0.match_expressions.1.values.1110466152: "" => "minikube" (forces new resource)
      spec.0.persistent_volume_reclaim_policy:                                                      "Retain" => "Retain"
      spec.0.persistent_volume_source.#:                                                            "1" => "1"
      spec.0.persistent_volume_source.0.host_path.#:                                                "1" => "1"
      spec.0.persistent_volume_source.0.host_path.0.path:                                           "/mnt/local-volume" => "/mnt/local-volume"


Plan: 1 to add, 0 to change, 1 to destroy.

------------------------------------------------------------------------
```

Reply:
The node_selector_term is actually similar to the label selector in schema_label_selector.go

They are similar but not the same. It is used for selecting node for pod.
I see there is ForceNew in the label selector, so maybe better to keep it.

I have reverted the change of supporting more patch operation.
It should still work, the test pass with ReplaceOperation.